### PR TITLE
fix: dispatchBeforeRedirect

### PR DESCRIPTION
**Pull request by zaosoula**
Patch from: https://github.com/k1de/axios/commit/8b5cab4e318cb7cbecbfecdf0a6ff49b7e3cc70a
Fix: https://github.com/axios/axios/issues/5365

### CODE REVIEW

**Comment by ialexryan** [comment]
Hi @DigitalBrainJS! Sorry for the direct ping. Could you please consider merging this tiny PR? It fixes a breaking bug (#5365) that is [blocking](https://github.com/axios/axios/issues/6006#issuecomment-1810334234) myself and others from upgrading to 1.x. Thank you! 🙏🏻

File: test/unit/adapters/http.js [hunk 1]
```diff
@@ -368,8 +368,8 @@ describe('supports http with nodejs', function () {
     }).listen(4444, function () {
       axios.get('http://localhost:4444/', {
         maxRedirects: 3,
-        beforeRedirect: function (options) {
-          if (options.path === '/foo') {
+        beforeRedirect: function (options, responseDetails) {
+          if ([options.path, responseDetails.headers.location].every(item => item === '/foo')) {
```
Comment (DigitalBrainJS) [inline test/unit/adapters/http.js:372]: Maybe it's better to use two simple conditions?

**Review by DigitalBrainJS — CHANGES_REQUESTED** [review]

File: dist/node/axios.cjs
[the remark this answers is not in the document]
Comment (zaosoula) [reply dist/node/axios.cjs:1]: Done ✅

File: dist/node/axios.cjs.map
[the remark this answers is not in the document]
Comment (zaosoula) [reply dist/node/axios.cjs.map:1]: Done ✅

File: test/unit/adapters/http.js (hunk 1 above)
Comment (zaosoula) [reply test/unit/adapters/http.js]: Done ✅

**Review by DigitalBrainJS — APPROVED** [review]

**Pull request merged**